File: app/src/main/java/com/hgianastasio/biblioulbrav2/views/fragments/BaseFragment.java
package com.hgianastasio.biblioulbrav2.views.fragments;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.hgianastasio.biblioulbrav2.di.components.ActivityComponent;
import com.hgianastasio.biblioulbrav2.views.activities.base.BaseActivity;

import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by heitor_12 on 09/05/17.
 */

public abstract class BaseFragment extends Fragment {
    protected Unbinder unbinder;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(getLayoutID(),container,false);
        preBind();
        unbinder = ButterKnife.bind(this,view);
        postBind();
        return view;
    }

    protected void preBind(){}

    protected void postBind(){}

    @LayoutRes
    protected abstract int getLayoutID();

    public abstract String getFragTag();

    public abstract String getTitle();

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }

    ActivityComponent getComponent(){
        return ((BaseActivity)getActivity()).getActivityComponent();
    }
}

File: app/src/main/java/com/hgianastasio/biblioulbrav2/views/fragments/BaseFragment.kt
package com.hgianastasio.biblioulbrav2.views.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment
import butterknife.ButterKnife
import butterknife.Unbinder
import com.hgianastasio.biblioulbrav2.di.components.ActivityComponent
import com.hgianastasio.biblioulbrav2.views.activities.base.BaseActivity

/**
 * Created by heitor_12 on 09/05/17.
 */
abstract class BaseFragment : Fragment() {
    protected var unbinder: Unbinder? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(layoutID, container, false)
        preBind()
        unbinder = ButterKnife.bind(this, view)
        postBind()
        return view
    }

    protected fun preBind() {}
    protected open fun postBind() {}

    @get:LayoutRes
    protected abstract val layoutID: Int
    abstract val fragTag: String
    abstract val title: String
    override fun onDestroyView() {
        super.onDestroyView()
        unbinder!!.unbind()
    }

    val component: ActivityComponent?
        get() = (activity as? BaseActivity)?.activityComponent
}